packages/moonstone/internal/DateTimeDecorator/DateTimeDecorator.js

```diff
@@ -211,12 +213,12 @@ const DateTimeDecorator = hoc((config, Wrapped) => {
 			let order = defaultOrder;
 
 			// Guard for isomorphic builds
-			if (this.i18nContext) {
+			if (typeof window !== 'undefined' && i18n) {
 				if (value) {
-					label = this.i18nContext.formatter.format(value);
+					label = i18n().formatter.format(value);
```

Comment:
The `i18n()` callback has some cost to it so we should minimize invoking it. At a minimum, a single call here which stores the return value would be an improvement. Also, I think it'd be a good candidate for memoization based on the locale prop which would give us comparable performance to the current implementation, I expect.